ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersPage.tsx

```diff
@@ -1,44 +1,56 @@
-import React from 'react';
-import { useHistory, useParams } from 'react-router-dom';
-import { Button, PageSection } from '@patternfly/react-core';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Divider, PageSection, Title } from '@patternfly/react-core';
 
-import { complianceEnhancedCoveragePath } from 'routePaths';
+import PageTitle from 'Components/PageTitle';
+import useRestQuery from 'hooks/useRestQuery';
+import { getComplianceClusterStats } from 'services/ComplianceResultsStatsService';
+import { getTableUIState } from 'utils/getTableUIState';
 
-import CoveragesToggleGroup from './CoveragesToggleGroup';
 import CoveragesPageHeader from './CoveragesPageHeader';
+import CoveragesToggleGroup from './CoveragesToggleGroup';
+import ProfileClustersTable from './ProfileClustersTable';
 
 function ProfileClustersPage() {
-    const history = useHistory();
     const { profileName } = useParams();
+    const [currentDatetime, setCurrentDatetime] = useState<Date>(new Date());
+
+    const fetchProfileClusters = useCallback(
+        () => getComplianceClusterStats(profileName),
```

Comment:
Here is an observation, not necessarily a recommendation.
* On one side of the coin `useState<Date>(new Date())` might bend the React rule of no side effects, because it has a different value on every call, unlike `''` or `false` or `null`.
* On other side of the coin `setCurrentDatetime(new Date());` is explicitly in `useEffect` hook.

An alternative might call a wrapper function for `getComplianceClusterStats` that can return `{ currentDatetime: Date, profileClusters: ListComplianceClusterOverallStatsResponse }` via `new Date()` in a `then` block.

Reply:
I don't think that `useState<Date>(new Date())` will necessarily cause side effects. But the `new Date()` expression will be evaluated on every render, which would produce a new `Date` object. That might be unnecessary. There is a trick that will allow us to only evaluate it once though. You can do the following:

```
const [currentDatetime, setCurrentDatetime] = useState(() => new Date());
```

Also, I don't think you'll need to add the `<Date>` since Typescript will be able to determine the type based on the initial state.

Reply:
I like Mark's suggestion because the date would be tightly couple with `profileClusters` data being fetched. We ran into a typescript issue when trying to implement it though. _We_ can confidently say date would never be displayed if data isn't fetched but typescript doesn't know that. 

@sachaudh can you think of an instance where date would be re-evaluated without `profileClusters` being fetched? That would be an issue but I tried to think through edge cases and don't think that should happen.